### DISCUSSION TITLE
Commited Changes

### DISCUSSION THREAD

**Discussion by CreateSimpleWays** [community/community#22682] [New to GitHub]
<p>Hi!</p>
<p>I’m a new user of github and I try to understand or to know whether a commit saves my files even when my laptop would break down. It says commiting stores my files to the local repository, but this “cloud” is a bit hard to imagine for me. Can I call my last commit with a new laptop? If so, I would not wait too long with pushing my files at my project…  Thanks a lot for helping.</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="CreateSimpleWays" data-post="1" data-topic="129362">
<div class="title">
<div class="quote-controls"></div>
 CreateSimpleWays:</div>
<blockquote>
<p>If so, I would not wait too long with pushing my files at my project…</p>
</blockquote>
</aside>
<p>That’s exactly what you should do.</p>
<p>To explain a bit more: <code>git commit</code> creates a commit on the current branch in your local copy of the repository (stored in the <code>.git/</code> directory). This lets you work with history locally, e.g. compare changes or go back to a specific version of your code. That protects you against mistakes, but not against your hard drive breaking. You can backup a local repository like any other files.</p>
<p><code>git push</code> sends the commits on your current branch (or possibly another, depending on command line options) to a remote repository (e.g. on Github). If your laptop breaks and you get a new one, you can then clone the remote repository and you’ll have those commits that have been pushed to the remote repository.</p>
<p>One side note: You can <a href="https://git-scm.com/book/en/v2/Git-Branching-Branches-in-a-Nutshell">use branches</a> to work on things that aren’t ready for the main branch of your repository yet, or that you just want to try out. You can push all your branches separately.</p>

**Comment by CreateSimpleWays** [comment, marked as answer]
<p>Thank you very much for explaining me carefully. Question well answered.</p>
